ospf6d/ospf6_route.c

```diff
@@ -292,28 +292,28 @@ void ospf6_add_nexthop(struct list *nh_list, int ifindex, struct in6_addr *addr)
 
 void ospf6_route_zebra_copy_nexthops(struct ospf6_route *route,
 				     struct zapi_nexthop nexthops[],
-				     int entries)
+				     int entries, vrf_id_t vrf_id)
 {
 	struct ospf6_nexthop *nh;
 	struct listnode *node;
-	struct interface *ifp;
 	char buf[64];
 	int i;
 
 	if (route) {
 		i = 0;
 		for (ALL_LIST_ELEMENTS_RO(route->nh_list, node, nh)) {
-			ifp = if_lookup_by_index_all_vrf(nh->ifindex);
 			if (IS_OSPF6_DEBUG_ZEBRA(SEND)) {
+				const char *ifname;
 				inet_ntop(AF_INET6, &nh->address, buf,
 					  sizeof(buf));
+				ifname = ifindex2ifname(nh->ifindex, vrf_id);
```

Comment:
shouldn't the vrf be specified by the nh->vrf_id?

Reply:
I just returned it to how it was done before "vrf transformation changes" to fix the crash.

I think you have the right idea, but ospf6_nexthop doesn't have the vrf_id field at the moment. I suppose there will be more commits for actual VRF support in ospf6d and the field must be added. I can add a TODO comment.